src/app/views/onboarding/step2.coffee

```diff
@@ -18,8 +18,17 @@ angular.module 'mnoEnterpriseAngular'
 
     # Select or deselect an app
     vm.toggleApp = (app) ->
-      # User cannot add more apps
-      return if vm.maxAppsSelected && !app.checked
+      # Add conflictingApp attribute to marketplace apps
+      names = _.map(app.subcategories, 'name')
+      marketplaceApps = _.filter(vm.marketplace.apps, (marketApp) -> marketApp != app)
```

Comment:
I'd extract these loops in a separate method to help with readability.
I'd also try to reduce the number of iterations.

Something like:
```coffeescript
# Add conflictingApp attribute to marketplace apps
refreshConflictingApp = (app) ->
  # List of non multi_instantiable categories
  subCats = _.map(_.filter(app.subcategories, 'multi_instantiable', false), 'name')

  for marketApp in vm.marketplace.apps
    continue if marketApp == app

    if _.some(marketApp.subcategories, (subCat) -> not subCat.multi_instantiable and subCat.name in subCats)
      marketApp.conflictingApp = app
```